yal/pushmessages_optin.py

```diff
@@ -0,0 +1,122 @@
+import logging
+
+from vaccine.base_application import BaseApplication
+from vaccine.states import Choice, WhatsAppButtonState
+from vaccine.utils import get_display_choices
+from yal import rapidpro
+from yal.askaquestion import Application as AaqApplication
+from yal.mainmenu import Application as MainMenuApplication
+from yal.utils import get_generic_error, normalise_phonenumber
+
+logger = logging.getLogger(__name__)
+
+
+class Application(BaseApplication):
+    START_STATE = "state_start_pushmessage_optin"
+
+    async def state_start_pushmessage_optin(self):
+        choices = [
+            Choice("yes", "Yes, please!"),
+            Choice("no", "No thanks"),
+        ]
+        question = self._(
+            "\n".join(
+                [
+                    "No stress — whenever you're ready I'm here.",
+                    "",
+                    "If you'd like, I can also send you notifications once a day with "
+                    "relevant info that I've put together just for you.",
+                    "",
+                    "Would you like to get notifications?",
+                    "",
+                    "💡You can turn the notifications off at any time.",
+                    "",
+                ]
+            )
+        )
+        return WhatsAppButtonState(
+            self,
+            question=question,
+            choices=choices,
+            error=self._(get_generic_error()),
+            next={
+                "yes": "state_pushmessage_optin_yes_submit",
+                "no": "state_pushmessage_optin_no_submit",
+            },
+        )
+
+    async def state_pushmessage_optin_no_submit(self):
+        msisdn = normalise_phonenumber(self.inbound.from_addr)
+        whatsapp_id = msisdn.lstrip(" + ")
+        data = {
+            "opted_in": "False",
```

Comment:
@rudigiesler we have a "Opt out" flow, which resets all the notification timers, and now we've implemented this, I have no idea if these two things are meant to have the same functionality and if push messages refers to the scheduled notifications or just the surveys, etc, and what that entails for the service. Please advise on whether there is a better naming convention that I can use in this section

Reply:
Yeah maybe we can be more explicit about what it is for? `segmentation_opt_in` or `push_message_opt_in` or something like that? That way we're explicit that this is only for the new push message things, and not for any of the notifications/reengagement messages.

I think we probably want users to be able to opt out of one type but keep receiving the other. But also a place where you can opt out of everything?

Reply:
Ok cool :) I will update the opt out flow to include these :)